moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -391,11 +398,11 @@ void RobotTrajectory::setRobotTrajectoryMsg(const robot_state::RobotState& refer
         Eigen::Isometry3d t = tf2::transformToEigen(trajectory.multi_dof_joint_trajectory.points[i].transforms[j]);
         st->setJointPositions(trajectory.multi_dof_joint_trajectory.joint_names[j], t);
       }
-      this_time_stamp = trajectory.multi_dof_joint_trajectory.header.stamp +
-                        trajectory.multi_dof_joint_trajectory.points[i].time_from_start;
+      this_time_stamp = rclcpp::Time(trajectory.multi_dof_joint_trajectory.header.stamp) +
+                        rclcpp::Duration(trajectory.multi_dof_joint_trajectory.points[i].time_from_start.sec,trajectory.multi_dof_joint_trajectory.points[i].time_from_start.nanosec);
     }
 
-    addSuffixWayPoint(st, (this_time_stamp - last_time_stamp).toSec());
+    addSuffixWayPoint(st, (this_time_stamp.seconds() - last_time_stamp.seconds()));
```

Comment:
Same point about subtracting first then converting to seconds later.

Reply:
e6e7a0b645d756c2c406a526cc51ae4fbbf4bd3d